lti_consumer/lti_1p3/consumer.py

```diff
@@ -586,8 +640,42 @@ def generate_launch_request(
                 )
             }
 
-        # Call LTI Launch if Deep Linking is not
-        # set up or this isn't a Deep Link Launch
+        if self.proctoring and lti_message_hint in ['LtiStartProctoring', 'LtiEndAssessment']:
```

Comment:
generate_launch_request should turn into just a router to _generate_deep_linking and _generate_proctoring or similar names, this function now has too many different concerns